### ISSUE TITLE
IPFS link taking long

### ISSUE THREAD

**Issue by dgrmunch** [dev-bootcamp-2019/final-project-hochbruj#1]
Hey! How are u doing?
Just some feedback. I have been waiting for 20 minutes and the link https://ipfs.io/ipfs/QmQT59MDoAw1GXTNK1wtX4ixVBwq91V4hiVwVx2ssxWmiD/ still does not show anything.
Is it pined? Besides that, the dApp looks great (and very useful)! :)

**Comment by hochbruj** [comment]
Thanks for the feedback! Yes, I did pin it, but the first time I
accessed it via the gateway it took a long time to load. I don't know
exactly how it is cached in the gateways. If I access the link from my
computers/browsers it is super fast. Maybe you can try again?

I like your dApp, but I haven't tested every functionality yet. Which
truffle version did you use? I am getting errors when I run your truffle
tests, but this could be because I am still using truffle v.4.





Am Mo., 28. Jan. 2019 um 20:07 Uhr schrieb Diego González Rodríguez <
notifications@github.com>:

**Comment by hochbruj** [comment]
I have updated to truffle v.5. I can compile and migrate your contacts and
interact with on local development app server.
But truffle test fails with error message:
Error: Could not find artifacts for
\C\Users\Jan\consensys\final-project-dgrmunch\contracts\CredentialsConstrained.sol
from any sources

Am Di., 29. Jan. 2019 um 11:11 Uhr schrieb Jan Hochbrückner <
jhochbrueckner@googlemail.com>:

**Comment by dgrmunch** [comment]
Hi @hochbruj!

No problem ;)
It looks like something is wrong with the cache of the gateway so probably not a bit issue. If you want I can try again while you are online to see if it works better. Just for the future, a good option can be to use another online service for IPFS pinning or running your own IPFS instance in a server.

Don't worry anyway. I have already evaluated your project and I have included the extra points of IPFS because, despite the problem with the gateway, you have taken the effort of uploading it and it is obviously a minor problem.

Said that, congratulations on the project. By the way, I hope you release a production version! It can be a very useful tool for the community!

Regarding the tests, probably you are facing a versioning problem.

I have executed the "truffle test" command in the root path of my project (at commit efb6a68) and it works well with Truffle v5.0.2 (core: 5.0.2) & Solidity v0.5.0 (solc-js) on OS X:

  ProofOfLifeProxyTest
    ✓ testContractOwner (102ms)
    ✓ testContractOpenToEveryUser (111ms)
    ✓ testAuthorizedUsers (183ms)
    ✓ testLockMethods (187ms)
    ✓ testCertificationMethods (171ms)

  ProofOfLifeTest
    ✓ testContractOwner (95ms)
    ✓ testContractOpenToEveryUser (182ms)
    ✓ testAuthorizedUsers (285ms)
    ✓ testLockedStatus (195ms)
    ✓ testLockedMethods (242ms)
    ✓ testLockedMethodsForParentContract (319ms)


  11 passing (30s)

Can you checkout to that commit and "ls" the contents of <root_path>/contracts/?
The contract should be there.

Best!

**Comment by hochbruj** [comment]
Thanks for the IPFS points even though it doesn't work :-)

I am running Truffle v5.0.2 (core: 5.0.2) and the behavior is strange
because compilation and migration of your project working fine and
CredentialsConstrained.sol is in the folder. Maybe it is a bug with the
solidity tests on truffle for windows (which I am running). If nobody else
of the group gets this error I will just give you the points.

I never thought about releasing a production version. Are you doing it with
your dApp?

BTW what do think about having a virtual meeting of our group after the
evaluation period? We could talk about our experiences and give some
feedback.


Am Di., 29. Jan. 2019 um 23:43 Uhr schrieb Diego González Rodríguez <
notifications@github.com>:

**Comment by dgrmunch** [comment]
Yes, the idea is that I will use a modified version of the contracts used in this dApp within the context of the company I work for. I will reuse probably parts of the dApp, but the smart contracts will be also consumed by a Java application with web3j.

I would love to have that meeting after the evaluation period too ;)
I think we can nurture a nice community to help each other in the future sharing knowledge and advances.
